src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBase.java

```diff
@@ -0,0 +1,168 @@
+package com.cloudbees.jenkins.plugins.sshagent;
+
+import com.trilead.ssh2.crypto.Base64;
+import com.trilead.ssh2.packets.TypesWriter;
+import org.apache.sshd.SshServer;
+import org.apache.sshd.server.PublickeyAuthenticator;
+import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
+import org.apache.sshd.server.session.ServerSession;
+import org.apache.sshd.server.shell.ProcessShellFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.security.PublicKey;
+import java.security.interfaces.DSAParams;
+import java.security.interfaces.DSAPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.util.EnumSet;
+
+public class SSHAgentBase {
```

Comment:
(Though beware that I recall some problem with `RestartableJenkinsRule` not handling `@Before`/`@After`. Not sure if that also applies to extra `@Rule`s.)